### PR TITLE
success callback of createImg updated

### LINKED ISSUE #4103: drop() callback not working with images
#### Nature of issue?

- [x] Found a bug

#### Most appropriate sub-area of p5.js?

- [x] DOM

#### Which platform were you using when you encountered this?

- [x] Desktop/Laptop

#### Details about the bug: 

- p5.js version:v0.10.2 October 14, 2019
- Web browser and version: 77.0.3865.120 (Official Build) (64-bit)
- Operating System: MacOSX Catalina

- Steps to reproduce this:
Trying to run this example: https://p5js.org/examples/dom-drop.html

```javascript
function setup() {
  // create canvas
  const c = createCanvas(710, 400);
  background(100);
  // Add an event for when a file is dropped onto the canvas
  c.drop(gotFile);
}

function draw() {
  fill(255);
  noStroke();
  textSize(24);
  textAlign(CENTER);
  text('Drag an image file onto the canvas.', width / 2, height / 2);
  noLoop();
}

function gotFile(file) {
  // If it's an image file
  if (file.type === 'image') {
    // Create an image DOM element but don't show it
    const img = createImg(file.data).hide();
    // Draw the image onto the canvas
    image(img, 0, 0, width, height);
  } else {
    console.log('Not an image file!');
  }
}
```
Everything is working, but the image is not placed on the canvas.

I think is something related to the fact that `image()` is called before the image is fully loaded, indeed if I change the code in this way it works:

```javascript
let img;

function setup() {
  // create canvas
  const c = createCanvas(710, 400);
  background(100);
  // Add an event for when a file is dropped onto the canvas
  c.drop(gotFile);
}

function draw() {
  fill(255);
  noStroke();
  textSize(24);
  textAlign(CENTER);
  text('Drag an image file onto the canvas.', width / 2, height / 2);
  noLoop();
  if(img){
    image(img, 0, 0, width, height);
  }
}

function gotFile(file) {
  // If it's an image file
  if (file.type === 'image') {
    // Create an image DOM element but don't show it
    img = createImg(file.data).hide();
    // Draw the image onto the canvas
    // image(img, 0, 0, width, height);
  } else {
    console.log('Not an image file!');
  }
}
```
I don't know if I'm missing something, if is an intended feature or a bug.

The `createFileInput()` function is anyway working.

### PR DESCRIPTION
**Pull request by SachinVarghese**
Resolves #4103

 Changes: 

- The successCallback of createImg() updated to pass back the image.
- The alt property of the image element only created if a string argument is passed.

 Screenshots of the change: 

#### PR Checklist

- [x] `npm run lint` passes
- [ ] [Inline documentation] is included / updated
- [ ] [Unit tests] are included / updated
- [ ] [Benchmarks] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/master/developer_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/master/developer_docs#unit-tests
[Benchmarks]: https://github.com/processing/p5.js/blob/master/developer_docs/benchmarking_p5.md

### CODE REVIEW

File: src/dom/dom.js [hunk 1]
```diff
@@ -412,16 +412,21 @@ p5.prototype.createImg = function() {
...
-    var last = args[args.length - 1];
-    if (typeof last === 'function') last();
-  });
+  elt.addEventListener(
+    'load',
+    function() {
+      self.width = elt.offsetWidth || elt.width;
+      self.height = elt.offsetHeight || elt.height;
+      var last = args[args.length - 1];
+      if (typeof last === 'function') last((self = addElement(elt, this)));
```
Comment (outofambit) [inline src/dom/dom.js]: Having `addElement(elt, this)` here as well as on the last line of the `createImg` function means we'll actually create two elements with this image source. i think we could instead set `self = addElement(elt, this)` once, right before we create this event listener. Then could pass `self` to the callback function, as well as `return` it. What do you think?

File: src/dom/dom.js [hunk 2]
```diff
@@ -412,16 +412,21 @@ p5.prototype.createImg = function() {
   var elt = document.createElement('img');
   var args = arguments;
   var self;
-  elt.alt = args[1];
+  if (args.length > 1 && typeof args[1] === 'string') {
+    elt.alt = args[1];
```
Comment (outofambit) [inline src/dom/dom.js]: i like this! 💯

**Comment by SachinVarghese** [comment]
@outofambit Thanks for your suggestions. I could agree with all of them and have made relevant changes. Let me know if this looks good. Also, I would be glad to contribute more and learn with this awesome project you guys are working on. Let me know how I can connect with the core team and the p5 JS community in general.

File: src/dom/dom.js (hunk 1 above)
Comment (SachinVarghese) [reply src/dom/dom.js]: Totally agreed on that!

**Comment by SachinVarghese** [comment]
Hi @outofambit Can you please review these PR updates?

File: src/dom/dom.js [hunk 3]
```diff
@@ -384,46 +384,44 @@ tags.forEach(function(tag) {
 });
 
 /**
- * Creates an &lt;img&gt; element in the DOM with given src and
- * alternate text.
+ * Creates an &lt;img&gt; element in the DOM with given source
  *
  * @method createImg
  * @param  {String} src src path or url for image
- * @param  {String} alt <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img#Attributes">alternate text</a> to be used if image does not load. You can use also an empty string (`""`) if that an image is not intended to be viewed.
```
Comment (outofambit) [inline src/dom/dom.js]: I think we'd like to keep `alt` as a required argument for the shorter version of this, to encourage people to use alt text for their images

**Pull request merged**